### PR TITLE
Feature: Login and signup modal

### PR DESCRIPTION
**Pull request by nguyenfamj**

### CODE REVIEW

File: src/modules/checkout/components/submit-button/index.tsx
Comment (Thangphan0102) [inline src/modules/checkout/components/submit-button/index.tsx:1]: You may consider to move this button component to modules/common/components.

File: src/modules/account/components/login/index.tsx [hunk 1]
```diff
@@ -1,57 +1,83 @@
-import { useFormState } from "react-dom"
+import { useFormState } from 'react-dom';
 
-import { LOGIN_VIEW } from "@modules/account/templates/login-template"
-import Input from "@modules/common/components/input"
-import { logCustomerIn } from "@modules/account/actions"
-import ErrorMessage from "@modules/checkout/components/error-message"
-import { SubmitButton } from "@modules/checkout/components/submit-button"
+import { LOGIN_VIEW } from '@modules/account/templates/login-template';
```
Comment (Thangphan0102) [inline src/modules/account/components/login/index.tsx:3]: This LOGIN_VIEW should be imported from src/modules/account/components/login-modal/index.tsx.

File: src/modules/account/components/register/index.tsx [hunk 2]
```diff
@@ -1,92 +1,107 @@
...
+import { useFormState } from 'react-dom';
 
-import Input from "@modules/common/components/input"
-import { LOGIN_VIEW } from "@modules/account/templates/login-template"
-import { signUp } from "@modules/account/actions"
-import ErrorMessage from "@modules/checkout/components/error-message"
-import { SubmitButton } from "@modules/checkout/components/submit-button"
-import LocalizedClientLink from "@modules/common/components/localized-client-link"
+import Input from '@modules/common/components/input';
+import { LOGIN_VIEW } from '@modules/account/templates/login-template';
```
Comment (Thangphan0102) [inline src/modules/account/components/register/index.tsx:6]: This LOGIN_VIEW should be imported from src/modules/account/components/login-modal/index.tsx.

File: src/modules/account/components/login-modal/index.tsx [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+      ref={modalRef}
+    >
+      <div className="modal-box">
+        <form method="dialog">
+          <button className="btn btn-circle btn-ghost absolute right-3 top-3 text-lg rounded-full bg-neutral hover:bg-gray-300">
+            ✕
+          </button>
+        </form>
+        <div className="mt-12 px-4 py-2">
+          {currentView === 'login' ? (
```
Comment (Thangphan0102) [inline src/modules/account/components/login-modal/index.tsx:30]: Consider to use the typed enum here instead of "login".

Comment (nguyenfamj) [reply src/modules/account/components/login-modal/index.tsx:30]: Valid!

**Review by Thangphan0102 — APPROVED** [review]

**Pull request merged**